ed25519.go

```diff
@@ -0,0 +1,87 @@
+package jwt
+
+import (
+	"errors"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	ErrEd25519Verification = errors.New("ed25519: verification error")
+)
+
+// Implements the EdDSA family
+// Expects *ed25519.PrivateKey for signing and *ed25519.PublicKey for verification
+type SigningMethodEd25519 struct{}
+
+// Specific instance for EdDSA
+var (
+	SigningMethodEdDSA *SigningMethodEd25519
+)
+
+func init() {
+	SigningMethodEdDSA = &SigningMethodEd25519{}
+	RegisterSigningMethod(SigningMethodEdDSA.Alg(), func() SigningMethod {
+		return SigningMethodEdDSA
+	})
+}
+
+func (m *SigningMethodEd25519) Alg() string {
+	return "EdDSA"
+}
+
+// Implements the Verify method from SigningMethod
+// For this verify method, key must be an Ed25519.PublicKey struct
+func (m *SigningMethodEd25519) Verify(signingString, signature string, key interface{}) error {
+	var err error
+
+	// Decode the signature
+	var sig []byte
+	if sig, err = DecodeSegment(signature); err != nil {
+		return err
+	}
+
+	// Get the key
+	var ed25519Key ed25519.PublicKey
+
+	switch k := key.(type) {
+	case *ed25519.PublicKey:
+		ed25519Key = *k
+	default:
+		return ErrInvalidKeyType
+	}
```

Comment:
Since only one type is checked I'd prefer:
```go
	var (
		ed25519Key *ed25519.PublicKey
		ok bool
	)
	if ed25519Key, ok = key.(*ed25519.PublicKey); !ok {
		return ErrInvalidKeyType
	}
```

But might be just my personal preference...